packages/botonic-plugin-flow-builder/src/action.tsx

```diff
@@ -18,31 +19,33 @@ export class FlowBuilderAction extends React.Component<FlowBuilderActionProps> {
     const flowBuilderPlugin = getFlowBuilderPlugin(request.plugins)
     const locale = flowBuilderPlugin.getLocale(request.session)
     const payload = request.input.payload
-    let targetContentId: string | undefined = payload
+    let targetNode: HtNodeWithContent | string | undefined = payload
+
     if (!payload && request.session.is_first_interaction) {
-      targetContentId = await flowBuilderPlugin.getStartId()
+      targetNode = flowBuilderPlugin.cmsApi.getStartNode()
     }
-    if (!payload) {
-      const intentPayload = await flowBuilderPlugin.getPayloadByIntent(
+
+    if (!payload && request.input.data) {
+      const intentNode = flowBuilderPlugin.cmsApi.getNodeByIntent(
         request.input,
         locale
       )
-      if (intentPayload) targetContentId = intentPayload
-      const keywordPayload = await flowBuilderPlugin.getPayloadByKeyword(
-        request.input,
+      const keywordNode = flowBuilderPlugin.cmsApi.getNodeByKeyword(
+        request.input.data,
         locale
       )
-      if (keywordPayload) targetContentId = keywordPayload
+      targetNode = intentNode ?? keywordNode ?? targetNode
```

Comment:
in conjunction with the above comments
`targetNode = intentNode ?? keywordNode ?? startTargetNode ?? payload`

Reply:
Before doing any changes here, I think we should agree about the priority between the starting content, the keywords and the intents. 
The way the code was done (and it is also now), we're prioritizing the intents over the keywords over the starting content, is this what we really want? Shouldn't we prioritize the starting content instead?